### PR TITLE
[Replace Non-Blocking Usage] LPS-198858 

### PR DESCRIPTION
**Pull request by JanisZhang**
Replace non-blocking ServiceProxyFactory usages in class `ScreenNameValidatorFactory`, `ScreenNameGeneratorFactory`, `EmailAddressValidatorFactory`, `EmailAddressGeneratorFactory`. Thanks!

https://liferay.atlassian.net/browse/LPS-198858

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@JanisZhang No regressions found.

File: portal-impl/src/META-INF/security-spring.xml [hunk 1]
```diff
@@ -21,8 +21,6 @@
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultFullNameValidator" id="com.liferay.portal.kernel.security.auth.DefaultFullNameValidator" />
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultScreenNameGenerator" id="com.liferay.portal.kernel.security.auth.DefaultScreenNameGenerator" />
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultScreenNameValidator" id="com.liferay.portal.kernel.security.auth.DefaultScreenNameValidator" />
-	<bean class="com.liferay.portal.kernel.security.auth.LiberalEmailAddressValidator" id="com.liferay.portal.kernel.security.auth.LiberalEmailAddressValidator" />
```
Comment (dantewang) [inline portal-impl/src/META-INF/security-spring.xml:24]: @JanisZhang Do you mean that they are only used by HookHotDeployListener, so they don't need to be published as Spring beans?

Comment (JanisZhang) [reply portal-impl/src/META-INF/security-spring.xml:24]: @dantewang yes.

Comment (dantewang) [reply portal-impl/src/META-INF/security-spring.xml:24]: @JanisZhang It's something similar with the full name generator?

Comment (JanisZhang) [reply portal-impl/src/META-INF/security-spring.xml:24]: @dantewang
Yes, these classes should be defined in portal properties `users.screen.name.validator`,` users.email.address.validator`
and override the default one by HookHotDeployListener.

**Comment by dantewang** [comment]
ci:forward:force

cc @shuyangzhou

**Pull request closed without merging**